src/rosdiscover/observer/nodeinfo.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+import re
+from typing import Collection, Set
+
+import attr
+from loguru import logger
+from roswire import ROS1
+
+_GOAL = re.compile(r'(.*)/goal')
+_GOAL_FMT = re.compile(r'(.*)Goal')
+_CANCEL = re.compile(r'(.*)/cancel')
+_STATUS = re.compile(r'(.*)/status')
+_FEEDBACK = re.compile(r'(.*)/feedback')
+_FEEDBACK_FMT = re.compile(r'(.*)Feedback')
+_RESULT = re.compile(r'(.*)/result')
+_RESULT_FMT = re.compile(r'(.*)Result')
+
+
+@attr.s(slots=True, auto_attribs=True)
+class _Action:
+    """Class for action information."""
+    name: str
+    fmt: str
+
+
+@attr.s(slots=True, auto_attribs=True)
+class NodeInfo:
+    """Class for partial node information."""
+    name: str
+    publishers: Set[str] = attr.ib(default=set())
+    subscribers: Set[str] = attr.ib(default=set())
+    provides: Set[str] = attr.ib(default=set())
+
+    @classmethod
+    def filter_topics_for_action(cls,
+                                 ros: ROS1, goal_related_topics: Set[str],
```

Comment:
Minor nitpick: if splitting arguments across multiple lines, then sticking to one argument per line makes it easier to read.

Reply:
Done